File: tests/fixtures/ruby-simple/lib/controllers/order_controller.rb
# Handles order operations including payment processing.
require_relative "../payments/payment_service"
require_relative "../utils/logger"

module Controllers
  class OrderController
    def initialize(payment_service, logger)
      @payment_service = payment_service
      @logger = logger
    end

    def create_order(order_id, amount)
      @logger.info("Creating order: #{order_id}")
      result = @payment_service.process_payment(order_id, amount)
      @logger.info("Order created successfully") if result
      result
    end

    def cancel_order(order_id)
      @logger.info("Cancelling order: #{order_id}")
      @payment_service.refund_payment(order_id)
    end
  end
end

File: tests/fixtures/php-simple/src/Controllers/OrderController.php
<?php

namespace App\Controllers;

use App\Payments\PaymentService;
use App\Payments\PaymentServiceInterface;
use App\Utils\Logger;

/** Handles order operations including payment processing. */
class OrderController
{
    private PaymentServiceInterface $paymentService;
    private Logger $logger;

    public function __construct(PaymentServiceInterface $paymentService, Logger $logger)
    {
        $this->paymentService = $paymentService;
        $this->logger = $logger;
    }

    public function createOrder(string $orderId, float $amount): bool
    {
        $this->logger->info("Creating order: {$orderId}");
        $result = $this->paymentService->processPayment($orderId, $amount);
        if ($result) {
            $this->logger->info("Order created successfully");
        }
        return $result;
    }

    public function cancelOrder(string $orderId): void
    {
        $this->logger->info("Cancelling order: {$orderId}");
        $this->paymentService->refundPayment($orderId);
    }

    public static function create(): self
    {
        $logger = new Logger("OrderController");
        $service = new PaymentService($logger);
        return new self($service, $logger);
    }
}
